handler.go

```diff
@@ -320,7 +337,14 @@ func (h FilteredHttpRequestHandler) Handle(req *http.Request, metrics Metrics) (
 		}
 	}
 
-	resp, err := h.client.Do(req)
```

Comment:
Perhaps it's obvious, but why change from `*http.Client` to `HttpRequestHandler`? Only apparent difference is that you can pass in `metrics` now, but that isn't used.

Reply:
It's because now in the tests in `gateway_test.go`, I can substitute `struct MockHttpRequestHandler`, which doesn't actually send a request to the gateway but returns a canned response.